src/front-door/azext_front_door/commands.py

```diff
@@ -89,6 +89,9 @@ def load_command_table(self, _):
     #     g.show_command('show', 'get')
     #     g.generic_update_command('update', custom_func_name='update_fd_backend_pool')
 
+    with self.command_group('network front-door probe', frontdoor_sdk) as g:
+        g.custom_command('update', 'update_fd_health_probe_settings')
```

Comment:
Could we use `generic_update_command`

Reply:
Lot's of commands are commented out like `network front-door probe create\delete`. I'm the new maintainer for network part. Could you give me some context about these commands?

Reply:
We dont support create/delete/update on probe, routingrules, frontend-endpoint. Initial swagger incorrectly stated that these are supported, which we fixed recently for all versions.

Reply:
@pichandwork Hello. Could we use generic_update_command instead of custom_command?

Reply:
@MyronFanQiu  My understanding is generic_update_command calls a PUT on the sub resource(in this case HealthProbeSettings). But PUT is not supported on HealthProbeSettings REST API. To update any settings in frontdoor, GET all settings from frontdoor, change the required values(HealthProbeSettings etc), then call PUT on frontdoor resource. So, thats why i had to use a custom command.

Reply:
@pichandwork generic_update_command will call a GET method by design to get an instance. Then you can change the values. Then it will use a PUT method to return the instance to the server. The logic is almost same with what you write. So I think it's better to use `generic_update_command `. https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#generic-update-commands.
`generic_update_command` will expose some general arguments for user. User can set some fields by `--set` in case you don't expose them to user.

Reply:
@MyronFanQiu - Which instance do you refer to when you say GET method to get an **instance**? Frontdoor instance or HealthProbeSettings instance? if its later(HealthProbeSettings), then there is no GET and PUT supported on this.